scvi/inference/trainer.py

```diff
@@ -65,6 +66,7 @@ def __init__(self, model, gene_dataset, use_cuda=True, metrics_to_monitor=None,
         self.use_cuda = use_cuda and torch.cuda.is_available()
         if self.use_cuda:
             self.model.cuda()
+        self.all_dataset = self.create_posterior(self)
```

Comment:
I think this should just be `self.create_posterior()`. The first argument is automatically self, right?

Reply:
It'd be better to call `create_posterior` in `get_dataset_information`, and just make `all_dataset` a local variable, not a class variable. `create_posterior` is fast so better to call it near where `all_dataset` is used.

Reply:
Absolutely the additional self is a typing error

Reply:
This would work too but I thought it would be useful to have this as an attribute of trainer at the same level than the train_set and test_set attributes. Furthermore there might be situations where we want to draw information on the whole dataset which aren't the latent space or the imputed values so I thought having this attribute might be useful

Reply:
Either way is fine. I can sort of see the first point---that it should be on the same level as the posterior objects. Though it's not like all three posterior objects get created on three consecutive lines.

I'm not so receptive to the second, because I think adding functionality that *might* be useful in future tends to make the code less readable. It's harder to keep track of the state of a class variable than local variable.

Reply:
I'll make it a local one so we'll just have to change it if it revealed to be useful later